source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/FunctionHost/ServiceCollectionRegistrationTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using GreenEnergyHub.Charges.FunctionHost;
+using GreenEnergyHub.Charges.Tests.TestHelpers;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.FunctionHost
+{
+    [UnitTest]
+    public class ServiceCollectionRegistrationTests
+    {
+        public ServiceCollectionRegistrationTests(IConfiguration config)
+        {
+            FunctionHostEnvironmentSettingHelper.SetFunctionHostEnvironmentVariablesFromSampleSettingsFile(config);
+        }
+
+        [Fact]
+        public void WhenApplicationIsConfigured_ThenAllApplicationDependenciesMustBeRegistered()
+        {
+            // Arrange
+            var program = new ChargesFunctionApp();
+            var host = program.ConfigureApplication();
+            var functionAppDependencies = FindDependenciesForType(typeof(ChargesFunctionApp));
+
+            // Act
+            foreach (var dependency in functionAppDependencies)
+            {
+                var constructorParameters = GetConstructorParametersForDependency(dependency);
+
+                foreach (var constructorParameter in constructorParameters)
+                {
+                    // Assert
+                    host.Services.GetService(constructorParameter).Should().NotBeNull();
+                }
+            }
```

Comment:
Could also be done like this, to avoid nesting foreach:
```suggestion
            // Act
            IEnumerable<Type> constructorParameters = new List<Type>();
            foreach (var dependency in functionAppDependencies)
                constructorParameters = GetConstructorParametersForDependency(dependency);

            // Assert
            foreach (var constructorParameter in constructorParameters)
                host.Services.GetService(constructorParameter).Should().NotBeNull();
```
I will leave it to you to decide which is better.